api_app/pivots_manager/pivots/abuse_domain_to_abuse_ip.py

```diff
@@ -0,0 +1,36 @@
+import logging
+from typing import Any, Optional, Tuple
+
+from api_app.pivots_manager.classes import Pivot
+from api_app.playbooks_manager.models import PlaybookConfig
+
+logger = logging.getLogger(__name__)
+
+
+class AbuseDomainToAbuseIp(Pivot):
+    def should_run(self) -> Tuple[bool, Optional[str]]:
+        playbook = PlaybookConfig.objects.filter(name="Abuse_Domain").first()
+        self.value = None
+        result = False
+        if self._job.playbook_requested == playbook:
+            for x in self.related_reports:
+                if (
+                    x.status == self.report_model.Status.SUCCESS.value
+                    and len(x.report["resolutions"]) > 0
+                ):
+                    value = x.report["resolutions"][0]["data"]
+                    if value:
+                        self.value = value
+                        result = True
+                        break
+        return (
+            result,
+            f"Necessary reports{'' if result else ' do not'} have success status "
+            f"{'and' if result else 'or'} needed data",
+        )
```

Comment:
I would have just checked that one of the analyzers of the configuration
(you have 4 analyzers in the migration) was executed
```suggestion
    def should_run(self) -> Tuple[bool, Optional[str]]:
        if valid_report := self.related_reports.filter(status=self.report_model.Status.SUCCESS.value).first():
            self._value = valid_report.report["resolutions"] # etc
             result = True
```

Reply:
Moreover, if we think about this, this is nothing more than the `Compare` Pivot where the `should_run` function checks that `AT LEAST ONE` report is successfully executed (instead of having exactly one report).
I would generalize that, calling it something like `AnyCompare` or whatever, and use as `field_to_compare` the value `resolutions.0.data`